src/btree/bt_sync.c

```diff
@@ -256,17 +256,21 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 
 			if (walk == NULL)
 				break;
+			page = walk->page;
 
 			/*
-			 * Skip clean pages, but need to make sure maximum
-			 * transaction ID is always updated.
+			 * Skip clean pages, but track the most recent update
+			 * in the tree. This determines when the tree can
+			 * safely be discarded from cache.
 			 */
-			if (!__wt_page_is_modified(walk->page)) {
-				if (((mod = walk->page->modify) != NULL) &&
-				    mod->rec_max_txn > btree->rec_max_txn)
+			if ((mod = page->modify) == NULL)
+				continue;
+			if (!__wt_page_is_modified(page)) {
+				if (WT_TXNID_LT(btree->rec_max_txn,
+				    mod->rec_max_txn))
```

Comment:
Checking `(mod = page->modify) == NULL)` isn't necessary, it could just be:
```
/*
 * Skip clean pages, but track the most recent update
 * in the tree. This determines when the tree can
 * safely be discarded from cache.
 */
if (!__wt_page_is_modified(page)) {
        mod = page->modify;
        if (WT_TXNID_LT(btree->rec_max_txn,
            mod->rec_max_txn))
                btree->rec_max_txn = mod->rec_max_txn;
```